Content.Server/GameObjects/EntitySystems/EmitSoundSystem.cs

```diff
@@ -0,0 +1,99 @@
+            SubscribeLocalEvent<EmitSoundOnActivateComponent, ActivateInWorldEvent>((eUI, comp, args) => PlaySoundBasedOnMode(eUI, comp));
+        }
+
+        private void PlaySoundBasedOnMode(EntityUid uid, BaseEmitSoundComponent component)
+        {
+            if (component == null)
+            {
+                return;
+            }
+#pragma warning disable CS8604 // Possible null reference argument.
+            switch (component._emitSoundMode)
+            {
+                case EmitSoundMode.Single:
+                    PlaySingleSound(component._soundName, component);
+                    break;
+                case EmitSoundMode.RandomFromCollection:
+                    PlayRandomSoundFromCollection(component);
+                    break;
+                case EmitSoundMode.Auto:
+                    if (!string.IsNullOrEmpty(component._soundCollectionName))
+                    {
+                        PlayRandomSoundFromCollection(component);
+                        break;
+                    }
+                    PlaySingleSound(component._soundName, component);
+                    break;
+                default:
+                    break;
+            }
+#pragma warning restore CS8604 // Possible null reference argument.
+        }
+
+        private void PlayRandomSoundFromCollection(BaseEmitSoundComponent component)
+        {
+            if (!string.IsNullOrWhiteSpace(component._soundCollectionName))
+            {
+                var file = SelectRandomSoundFromSoundCollection(component._soundCollectionName);
+                PlaySingleSound(file, component);
+            }
+        }
+
+        private string SelectRandomSoundFromSoundCollection(string soundCollectionName)
+        {
+            var soundCollection = _prototypeManager.Index<SoundCollectionPrototype>(soundCollectionName);
+            return _random.Pick(soundCollection.PickFiles);
+        }
+
+        private static void PlaySingleSound(string soundName, BaseEmitSoundComponent component)
+        {
+            if (string.IsNullOrWhiteSpace(soundName))
+            {
+                return;
+            }
+
+            if (component._pitchVariation > 0.0)
+            {
+                SoundSystem.Play(Filter.Pvs(component.Owner), soundName, component.Owner,
+                                 AudioHelpers.WithVariation(component._pitchVariation).WithVolume(-2f));
+            }
+            else if (component._semitoneVariation > 0)
+            {
+                SoundSystem.Play(Filter.Pvs(component.Owner), soundName, component.Owner,
+                                 AudioHelpers.WithSemitoneVariation(component._semitoneVariation).WithVolume(-2f));
+            }
+            else
+            {
+                SoundSystem.Play(Filter.Pvs(component.Owner), soundName, component.Owner,
+                                 AudioParams.Default.WithVolume(-2f));
+            }
+        }
+    }
+}
+
```

Comment:
As remie said remove the other functionality, just make it play a random soundcollection file and this should be much shorter. Also don't use semitonevariation.